tests/e2e/oci_registry_credentials_secret.go

```diff
@@ -0,0 +1 @@
+package e2e
```

Comment:
Is the secret defined in `pkg/test_samples/auth_secret.yaml` a sample secret for this test? If so, maybe it's worth adding a comment here that the secret should be used in this test.

Reply:
No, in the E2E scenario the secret needs to be created for the docker registry used in the k3d setup with something similar like:
```
kubectl create secret docker-registry my-private-registry-auth \
  --docker-server=http://my-private-registry:5000 \
  --docker-username=myuser \
  --docker-password=mypassword \
  --docker-email=dummy@example.com \
  -n kcp-system 
``` etc..
and then applied to the cluster, etc..